file_transfer_agent.go

```diff
@@ -746,6 +746,19 @@ func (sfa *snowflakeFileTransferAgent) uploadFilesParallel(fileMetas []*fileMeta
 			}
 			wg.Wait()
 
+			// append errors with no result associated to separate array
+			var errorMessages []string
+			for i, result := range results {
+				if result == nil {
+					errorMessages = append(errorMessages, errors[i].Error())
```

Comment:
Will it possible when result is nil, errors also nil?

Reply:
It should not be possible - the function `uploadOneFile` never returns nil, nil but I guess I can make the additional check just in case.